e2e/test/provisioning/ProvisioningE2ETests.cs

```diff
@@ -1030,8 +1030,8 @@ public static ProvisioningClientOptions CreateProvisioningClientOptionsFromName(
                                     iothubs,
                                     capabilities)
                                 .ConfigureAwait(false);
-                            Assert.IsTrue(symmetricKeyEnrollmentGroup.Attestation is SymmetricKeyAttestation);
-                            var symmetricKeyAttestation = (SymmetricKeyAttestation)symmetricKeyEnrollmentGroup.Attestation;
+                            symmetricKeyEnrollmentGroup.Attestation.Type.Should().Be(AttestationMechanismType.SymmetricKey);
```

Comment:
I suspect all this stuff got broke when we updated the json constructor stuff. Do these changes look right to you?